src/transformers/commands/add_new_model_like.py

```diff
@@ -402,12 +418,55 @@ def get_module_from_file(module_file: Union[str, os.PathLike]) -> str:
 _re_class_func = re.compile(r"^(?:class|def)\s+([^\s:\(]+)\s*(?:\(|\:)", flags=re.MULTILINE)
 
 
+def remove_attributes(obj, target_attr):
+    """Remove `target_attr` in `obj`."""
+    lines = obj.split(os.linesep)
+
+    target_idx = None
+    for idx, line in enumerate(lines):
+        # search for assignment
+        if line.lstrip().startswith(f"{target_attr} = "):
+            target_idx = idx
+            break
+        # search for function/method definition
+        elif line.lstrip().startswith(f"def {target_attr}("):
+            target_idx = idx
+            break
+
+    # target not found
+    if target_idx is None:
+        return obj
+
+    line = lines[target_idx]
+    indent_level = find_indent(line)
+    # forward pass to find the ending of the block (including empty lines)
+    parsed = parse_module_content("\n".join(lines[target_idx:]), indent_level)
+    if len(parsed) > 0:
+        parsed = parsed[0]
+        num_lines = len(parsed.split("\n"))
+        for idx in range(num_lines):
+            lines[target_idx + idx] = None
```

Comment:
I'm confused. Here it's just a while loop to find the first line with indent <= to the indent level. Why change `parse_module_content` for this?

Reply:
Basically, we want to have all lines:

 - empty line
 - or indent < target indent
 - or indent == target indent, but is a closing like `)`, `]`, `}`.

I didn't think about it much in advance - only found out these when I went directly to an existing method to modify it.

Reply:
Let me know if you still prefer not to modify `parse_module_content`. I can add a new method which is basic the same, but a bit simpler (i.e. no need to consider `is_valid_object` or `# Copied from")`

Reply:
This confuses me a lot because you use here (and modify above) `parse_module_content` to do something else entirely. It would be better to define another function that does what you want and just return the one object you want, even if it means a copy of the code.

Reply:
Sure, to avoid the complexity. I also don't like the name `parse_module_content` when I added the changes to it.

**But I wouldn't say the change does sth else entirely**. When when the target indent is `0`, it behaves the same as before, which extracts the blocks at the module level. If indent > 0, it extracts only the blocks of that indent level. It's just an extension, but really something else. The only issue to me is just the method naming is a bit confusing that way.

Reply:
@sgugger Done the change in [commit ](https://github.com/huggingface/transformers/pull/21999/commits/55546b6d1f0cb08de8299c5c92a07513461d8765)

If the changes in [add_new_model_like.py](src/transformers/commands/add_new_model_like.py) are all good, the remaining thing is just to move other exception cases to model specific test files.